docs/documentation/siddhi-4.0.md

```diff
@@ -363,7 +363,7 @@ Item|Description
 `@store`|This annotation is used to refer to the data source where the events for which aggregate values are to be calculated are stored.
 `define aggregation`|This specifies a unique name for the aggregation
 `group by`|The attribute by which the calculated aggregate values are grouped. Specifying an attribute to group by is optional. When an attribute is specified, the aggregate values are calculated for the required time periods per value for the specified attribute. If no attribute is specified, all the events are aggregated together.
-`aggregate by timestamp`|The time period for which the aggregate values are calculated.
+`aggregate by timestamp`|The time period for which the aggregate values are calculated. This is an optional parameter. If the time period is determined by an external timestamp (i.e., the timestamp specified as the value for the `_timestamp` attribute in the event), specific timestamps must be specified in the query with the `within` operator using supported formats (i.e., `<yyyy>-<MM>-<dd>`, `<HH>:<mm>:<ss>`, `<Z>` (if time is not in GMT), and `<yyyy>-<MM>-<dd> <HH>:<mm>:<ss>` (if time is in GMT). If the time period is to be determined based on the system time, you can specify the time duration for which the aggregate values should be calculated (e.g., `aggregate every sec...year` calculates aggregate values for the last second, minute, hour, day, month and year in a sliding manner.).
```

Comment:
This must change slightly as follows
i.e., the timestamp specified as the value for the 'timestamp' attribute in the event (not '_timestamp')
The 'within' details are not relevant here. We must include that information under 'Retrieving aggregate values' section. 
Mention that the values we get for 'timestamp' attribute could either be a long or a string. If it's a string, it must adhere to the formats `<yyyy>-<MM>-<dd>`, `<HH>:<mm>:<ss>`, `<Z>` (if time is not in GMT), and `<yyyy>-<MM>-<dd> <HH>:<mm>:<ss>` (if time is in GMT)
The calculation is not done in a sliding manner. It would be done in batches. For example, if we consider 'hour' aggregations, if we get one event at 2017-01-01 11:30:54 and another event at  2017-01-01 11:50:54, both would be processed as a batch for the 11th hour (all the events arriving within 2017-01-01 11:00:00 to 2017-01-01 12:00:00 would be processed as one batch)

Reply:
@suhothayan are we going to call this 'Time based aggregation'?